### PR TITLE
Add test workflow

### PR DESCRIPTION
**Pull request by tamiroh**
Currently, there is no workflow to check the quality of code when it is pushed to the repository. However, by setting up this, we can prevent code quality degradation and achieve safer development.

First, I incorporated eslint into the workflow, which requires minimal modification of existing code. I plan to add typescript compiler checks, etc. later.

### CODE REVIEW

File: src/gitManager.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export abstract class GitManager {
     */
     private simplify(tree: TreeItem[]): TreeItem[] {
         for (const node of tree) {
-            while (true) {
+            for (;;) {
```
Comment (tamiroh) [inline src/gitManager.ts:131]: ESLint's no-constant-condition rule does not allow `while (true)`. Therefore, I have replaced it with `for (;;)` instead, per the following issue. If you don't like it, please disable no-constant-condition.
https://github.com/eslint/eslint/issues/5477

File: src/simpleGit.ts [hunk 2]
```diff
@@ -50,9 +50,6 @@ export class SimpleGit extends GitManager {
                 process.env[key] = value;
             }
 
-            const debug = require('debug');
```
Comment (Vinzent03) [inline src/simpleGit.ts:53]: That's necessary for automated debug logs. https://github.com/steveukx/git-js/blob/main/docs/DEBUG-LOGGING-GUIDE.md

File: src/gitManager.ts (hunk 1 above)
Comment (Vinzent03) [reply src/gitManager.ts]: Yeah I prefer `while(true)` If disabled the rule and reverted the change. Thanks for pointing out!

File: src/simpleGit.ts (hunk 2 above)
Comment (tamiroh) [reply src/simpleGit.ts:53]: @Vinzent03 
OK👍 I added again
3723ca1

**Comment by Vinzent03** [comment]
LGTM, good idea to add some automated tests!

**Pull request merged**